packages/client/hmi-client/src/utils/stats.ts

```diff
@@ -61,3 +62,126 @@ export function quantile(numberList: number[], q: number) {
 
 	return lowerValue + (upperValue - lowerValue) * (pos - lowerIndex);
 }
+
+// Weighted Interval Score (WIS)
+// These are js ports of python functions specified in this issue: https://github.com/orgs/DARPA-ASKEM/projects/3/views/8?pane=issue&itemId=93848937&issue=DARPA-ASKEM%7Cterarium%7C6067
+// The original python code: https://github.com/adrian-lison/interval-scoring/blob/master/scoring.py
+
+const DEFAULT_ALPHA_QS = [
+	0.01, 0.025, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0.5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95,
+	0.975, 0.99
+];
+
+// Compute the estimated quantiles from a time-series dataset with many samples.
+export function computeQuantile(
+	summaryResult: Record<string, any>,
+	variableName: string,
+	alphas: number[] = DEFAULT_ALPHA_QS
+) {
+	const values = summaryResult.map((row) => row[variableName]);
+	const quantileValueMap: Record<number, number> = {};
+	alphas.forEach((q) => {
+		quantileValueMap[q] = quantile(values, q);
+	});
+	return quantileValueMap;
+}
+
+// Compute interval scores (1) for an array of observations and predicted intervals.
+// Either a dictionary with the respective (alpha/2) and (1-(alpha/2)) quantiles via q_dict needs to be
+// specified or the quantiles need to be specified via q_left and q_right.
+// This is a dumbed down version of the original function as it's just handling the case where the left and right quantiles are not specified, etc.
+export function getIntervalScore(
+	groundTruthObservations: Record<number, number>,
+	variableObservations: Record<number, number>,
+	alpha: number,
+	percent: boolean,
+	checkConsistency: boolean,
+	leftQuantile: number | null = null,
+	rightQuantile: number | null = null
```

Comment:
I do not recommend chaining optional parameters like this, it seems to suggest that both `leftQuantile` and `rightQuantile` are optional.  But if I only want to provide a `rightQuantile` I have to specify a leftQuantile (albeit sending in a null). 

From a modelling perspective, I think it is better to put optional fields into an object like:
```
{
  leftQuantile?: number,
  rightQuantile?: number,
 .... other stuff
}
```